src/status_im2/subs/communities.cljs

```diff
@@ -223,31 +207,62 @@
+                                                      :collapsed? (get collapsed-categories
+                                                                       category-id)
+                                                      :chats      [])))
+          chat                              {:name             name
+                                             :emoji            emoji
+                                             :unread-messages? (pos? unviewed-messages-count)
+                                             :position         position
+                                             :mentions-count   (or unviewed-mentions-count 0)
+                                             :locked?          (or (not joined)
+                                                                   (not can-post?))
+                                             :id               id}]
+      (update-in acc-with-category [category-id :chats] conj chat))))
+
 (re-frame/reg-sub
  :communities/categorized-channels
  (fn [[_ community-id]]
    [(re-frame/subscribe [:communities/community community-id])
-    (re-frame/subscribe [:chats/chats])])
- (fn [[{:keys [joined categories chats]} full-chats-data] [_ community-id]]
-   (reduce
-    (fn [acc [_ {:keys [name categoryID id emoji can-post?]}]]
-      (let [category                                                  (keyword
-                                                                       (get-in categories
-                                                                               [categoryID :name]
-                                                                               (i18n/label :t/none)))
-            {:keys [unviewed-messages-count unviewed-mentions-count]} (get full-chats-data
-                                                                           (str community-id id))]
-        (update acc
-                category
-                #(vec (conj %1 %2))
-                {:name             name
-                 :emoji            emoji
-                 :unread-messages? (pos? unviewed-messages-count)
-                 :mentions-count   (or unviewed-mentions-count 0)
-                 :locked?          (or (not joined) (not can-post?))
-                 :id               id})))
-    {}
-    chats)))
+    (re-frame/subscribe [:chats/chats])
+    (re-frame/subscribe [:communities/collapsed-categories-for-community community-id])])
+ (fn [[{:keys [joined categories chats]} full-chats-data collapsed-categories] [_ community-id]]
+   (let [reduce-fn (reduce-over-categories
+                    community-id
+                    joined
+                    categories
+                    collapsed-categories
+                    full-chats-data)
+         categories-and-chats
+         (->>
+           chats
+           (reduce
+            reduce-fn
+            {})
+           (sort-by (comp :position second))
```

Comment:
We could look into sorting in status-go as well, I need to have a look, this is not much work most likely as you are always sorting a limited number of elements (each category will have at most I guess 10 chats)